atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/PartitionByInvokeNameCheckerHelperTest.java

```diff
@@ -0,0 +1,206 @@
+            .add(TestEventUtils.invokeLock(6, PROCESS_2, LOCK_2))
+            .add(TestEventUtils.lockSuccess(7, PROCESS_2))
+            .build();
+
+    private static CheckerResult validResult = ImmutableCheckerResult.builder()
+            .valid(true)
+            .errors(new ArrayList<>())
+            .build();
+
+    /**
+     * This checker always fails, returning the entire list of events as the error list. This is useful to verify how
+     * PartitionByInvokeNameCheckerHelper manipulates the list. The value of the valid parameter is set to false purely
+     * for consistency reasons (since the list of errors is not empty).
+     */
+    private static Checker identityChecker = Mockito.mock(Checker.class);
+
+    static {
+        Mockito.when(identityChecker.check(Matchers.anyListOf(Event.class))).then(
+                list -> ImmutableCheckerResult.builder()
+                        .valid(false)
+                        .errors((List) list.getArguments()[0])
+                        .build()
+        );
+    }
+
+    @Test
+    public void universalSuccessCheckerShouldSucceedOnNoEvents() {
+        Checker universalChecker = Mockito.mock(Checker.class);
+        Mockito.when(universalChecker.check(Matchers.anyList())).thenReturn(validResult);
+
+        assertNoError(() -> universalChecker, ImmutableList.<Event>of());
+    }
+
+    @Test
+    public void universalSuccessCheckerShouldSucceed() {
+        Checker universalChecker = Mockito.mock(Checker.class);
+        Mockito.when(universalChecker.check(Matchers.anyList())).thenReturn(validResult);
+
+        assertNoError(() -> universalChecker, eventList);
+    }
+
+    @Test
+    public void infoEventsAreNotLost() {
+        InfoEvent infoEvent = TestEventUtils.createInfoEvent(0, PROCESS_1, RequestType.LOCK);
+
+        CheckerResult checkerResult = runPartitionChecker(() -> identityChecker,
+                ImmutableList.<Event>of(infoEvent));
+
+        assertThat(checkerResult.valid()).isFalse();
+        assertThat(checkerResult.errors()).containsExactly(infoEvent);
+    }
+
+    @Test
+    public void universalFailureCheckerShouldFail() {
+        InvokeEvent invokeEvent = TestEventUtils.invokeLock(0, PROCESS_1, LOCK_1);
+        OkEvent okEvent = TestEventUtils.lockSuccess(1, PROCESS_1);
+
+        CheckerResult checkerResult = runPartitionChecker(() -> identityChecker,
+                ImmutableList.of(invokeEvent, okEvent));
+
+        assertThat(checkerResult.valid()).isFalse();
+        assertThat(checkerResult.errors()).containsExactly(invokeEvent, okEvent);
+    }
+
+    @Test
+    public void partitioningOnlyReordersEvents() {
+        CheckerResult checkerResult = runPartitionChecker(() -> identityChecker, eventList);
+
+        assertThat(checkerResult.valid()).isFalse();
+        assertThat(checkerResult.errors().size()).isEqualTo(eventList.size());
+        assertThat(checkerResult.errors()).containsOnlyElementsOf(eventList);
```

Comment:
nit: `containsExactlyInAnyOrder(...)` would also take care of the size check for you.